src/perception/src/center_calculation/circunference_center_calculation.cpp

```diff
@@ -0,0 +1,59 @@
+#include <pcl/point_cloud.h>
+#include <pcl/point_types.h>
+
+#include <pcl/common/impl/centroid.hpp>
+#include <string>
+#include <center_calculation/circunferece_center_calculation.hpp>
+
+
+Eigen::Vector4f CircunferenceCenterCalculation::calculateCenter(pcl::PointCloud<pcl::PointXYZI>::Ptr point_cloud, const Plane& plane) const {
+    Eigen::Vector4f centroid;
+
+    // Get the centroid cone of the point_cloud;
+    pcl::compute3DCentroid(*point_cloud, centroid);
+
+    // Get the parallel plane that passes through that point
+
+    double d = - (plane.get_a() * centroid.x() + plane.get_b() * centroid.y() + plane.get_c() * centroid.z()); // d component of the plane
+
+    Plane middle_plane = Plane(plane.get_a(), plane.get_b(), plane.get_c(), d);
+
+    // Gets the 3 closests points to that plane
+    std::vector<pcl::PointXYZI> closest_points(3);
+    std::vector<double> distances(3, std::numeric_limits<double>::max());
+
+    for (auto& point : point_cloud->points) {
+        double distance = middle_plane.get_distance_to_point(point);
+
+        // Update distances/ closest point
+        for (int i = 0; i < 3; ++i) {
+            if (distance < distances[i]) {
+                for (int j = 2; j > i; --j) {
```

Comment:
Do you think you could explain this a bit better in a one liner or so? Or maybe don't but include a more detailed explanation in the latex later on